### PR TITLE
Improve skeleton proofreading and fix some shortcuts on MacOS

### LINKED ISSUE #7584: Ctrl+Click proofreading shortcut does not work on MacBook
Ctrl+Click is reserved on MacBooks so the user cannot access proofreading shortcut (I think it’s merge segments?)

reported by @valentin-pinkau

### LINKED ISSUE #7652: Zoom hotkey hint could be improved on macOS
## Detailed Description
After opening a dataset, the hotkey hint for zooming shows "Alt + {scroll mouse}":
Screenshot 2024-02-24 at 22 33 47

Macs don't have an "Alt" key though - it would be nice to detect if on macOS, and if so show "option + {scroll mouse}" instead.

## Context

* [ ] Specific to long-running jobs (set `jobsEnabled=true` in `application.conf`)
* [ ] Specific to webknossos.org (set `isDemoInstance=true` in `application.conf`)

### PR DESCRIPTION
**Pull request by philippotto**
- Enables the existing (legacy) shortcuts for deleting/creating edges for orthogonal and arbitrary mode. Now, they are not restricted to legacy mode anymore.
- Improves the shortcut indicator in the status bar.
- Allows creating trees and deleting nodes in arbitrary mode with the keyboard.
- Treats the Cmd (meta) key in MacOS as a synonym to ctrl everywhere. 
- Shows the ⌥ symbol instead of "alt" and Cmd instead of Ctrl in the status bar and context menu on MacOS.

### URL of deployed dev instance (used for testing):
- https://improveskeletonproofreading.webknossos.xyz/

### Steps to test:
- ortho mode:
  - create a tree with some nodes
  - delete an edge by holding ctrl+shift while clicking on the neighbor of the active node (the statusbar should indicate that action while pressing the modifiers)
  - merge the edge again by holding ctrl+alt (on some linux setups, shift also needs to be pressed to avoid that the default handler for moving a window is triggered)
  - the above should also work in legacy mode
  - test that node selection still works correctly (legacy mode should require shift for that)
- ortho mode
  - c should create a new tree
  - delete should delete the active node
  - clicking on a node should select it (sometimes the node is then not precisely visible, I don't really know how we could circumvent this)
  - as above, try out creating and deleting edges
- on mac:
  - test that https://github.com/scalableminds/webknossos/issues/7652 is fixed
  - test that https://github.com/scalableminds/webknossos/issues/7584 is fixed

### Issues:
- https://scm.slack.com/archives/C5AKLAV0B/p1705582675454289
- fixes #7584
- fixes #7652

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Removed dev-only changes like prints and application.conf edits
- [ ] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [ ] Needs datastore update after deployment

### CODE REVIEW

**Review by daniel-wer — COMMENTED** [review]
Nice improvements! Code looks great :+1: 

Notes from testing:
- [x]  Ctrl + Leftclick to delete meshes doesn't work (I suggested the fix during my review, I just want  to remember to retest this)
- [ ] From the internal feature request that spawned this PR, number (1) refers to easy skeleton merging and splitting using keyboard shortcuts similar to proofreading. Would it be possible to reuse the shortcuts used in proofreading mode or would this be a breaking change?
- [x] The keyboard shortcut overview is no longer accurate. For example, the arbitrary mode section doesn't show that nodes can be deleted or edges can be removed/created.
- [x] In flight/oblique mode, the active node is not highlighted making it hard to know what's going to happen if delete is pressed or an edge is created.
- [ ] Someone with a Mac should test the Mac-related fixes in this PR

File: frontend/javascripts/oxalis/controller/viewmodes/arbitrary_controller.tsx [hunk 1]
```diff
@@ -85,34 +87,53 @@ class ArbitraryController extends React.PureComponent<Props> {
...
-        scroll: this.scroll,
-        pinch: (delta: number) => {
-          if (delta < 0) {
-            Store.dispatch(zoomOutAction());
-          } else {
-            Store.dispatch(zoomInAction());
-          }
-        },
-      });
+        "arbitraryViewport",
```
[the remark this answers is not in the document]
Comment (philippotto) [reply frontend/javascripts/oxalis/controller/viewmodes/arbitrary_controller.tsx:135]: no, they are different (one is the element ID in the dom) and one is the viewport string (similar to PLANE_XY). nevertheless, we got a constant for that, which I inserted here now.

**Comment by philippotto** [comment]
@valentin-pinkau could you test the two mac-related items from the test plan? :)

> on mac:
> test that https://github.com/scalableminds/webknossos/issues/7652 is fixed
> test that https://github.com/scalableminds/webknossos/issues/7584 is fixed

**Comment by valentin-pinkau** [comment]
> @valentin-pinkau could you test the two mac-related items from the test plan? :)

Yes, both issues are fixed!

    test that https://github.com/scalableminds/webknossos/issues/7652 is fixed
    test that https://github.com/scalableminds/webknossos/issues/7584 is fixed


However, it would be great to show the mac user to use CMD click to split two segments.
Screenshot 2024-03-21 at 10 59 32

**Comment by philippotto** [comment]
> However, it would be great to show the mac user to use CMD click to split two segments.

Good point, done!

File: frontend/javascripts/oxalis/view/rendering_utils.ts [hunk 2]
```diff
@@ -51,19 +52,30 @@ export function renderToTexture(
...
+    ) {
+      const isArbitraryMode = constants.MODES_ARBITRARY.includes(
+        state.temporaryConfiguration.viewMode,
+      );
+      camera = camera.clone();
+      if (isArbitraryMode) {
+        // camera.far = state.userConfiguration.clippingDistanceArbitrary;
+        camera.near = ARBITRARY_CAM_DISTANCE - state.userConfiguration.clippingDistanceArbitrary;
+      } else {
+        camera.far = state.userConfiguration.clippingDistance;
```
[the remark this answers is not in the document]
Comment (daniel-wer) [reply frontend/javascripts/oxalis/view/rendering_utils.ts]: Thank you for your further investigations!

>  It seemed like regardless of what I did, I could always select nodes even when they weren't visible and I did stuff like setting near=far=0.

One more thing that just came to my mind is that I think we enlarge the nodes in the picking shader a bit to make it easier to click them - maybe this is done in z as well :thinking: 

However, we don't need to investigate this further as part of this PR. Did you find an existing issue or could you create one so we remember to have another look at this in the future.

**Review by daniel-wer — APPROVED** [review]
Mostly LGTM, but two things to consider :)

- [ ] The current active tool seems to remain active when switching to flight or oblique mode. So when the brush is active before switching, the mouse pointer is wrong or when the proofreading tool is active before, skeletons cannot be modified. This is, although the tools are not visible and cannot be switched, there. This doesn't necessarily need to be fixed in this PR as the bug likely existed before, but users might hit it more frequently now that skeleton modifications are allowed in flight mode.
- [ ] The casing of Shift and Ctrl in the context menu shortcuts is mixed now. Would be nice to align that.
  image

File: frontend/javascripts/oxalis/geometries/cube.ts [hunk 3]
```diff
@@ -52,15 +52,23 @@ class Cube {
       PLANE_XZ: new THREE.Line(new THREE.BufferGeometry(), this.getLineMaterial()),
       PLANE_YZ: new THREE.Line(new THREE.BufferGeometry(), this.getLineMaterial()),
     };
+    console.log("would set visible to false in", this);
+    this.setVisibility(false);
+    // for (const planeId of OrthoViewValuesWithoutTDView) {
+    //   this.crossSections[planeId].visible = false;
+    // }
+    // this.cube.visible = false;
```
[the remark this answers is not in the document]
Comment (philippotto) [reply frontend/javascripts/oxalis/geometries/cube.ts:60]: oh, i didn't intend to commit this :see_no_evil:

[the remark this answers is not in the document]
Comment (philippotto) [reply frontend/javascripts/oxalis/geometries/cube.ts:60]: line 56 is not a proper fix unfortunately. originally, I tried to fix the bbox bug when starting in arbitrary mode, but it turned out to be quite confusing why this appeared in the first place (the scene controller calls `stopPlaneMode` which should hide all meshes, too). I think, the initialization order is at fault, but I ran out of time fixing this properly.

File: frontend/javascripts/oxalis/view/rendering_utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (philippotto) [reply frontend/javascripts/oxalis/view/rendering_utils.ts]: > One more thing that just came to my mind is that I think we enlarge the nodes in the picking shader a bit to make it easier to click them - maybe this is done in z as well 🤔

Interesting idea! However, [here](https://registry.khronos.org/OpenGL-Refpages/gl4/html/gl_PointSize.xhtml#:~:text=If%20GL_PROGRAM_POINT_SIZE%20is%20enabled%2C%20gl_PointSize,ignored%20by%20the%20rasterization%20stage.) it is written that gl_PointSize "contains size of rasterized points, in pixels". So, I would assume that it only impacts the size in 2D as rasterization is explicitly mentioned.

> Did you find an existing issue or could you create one so we remember to have another look at this in the future.

I just created #7725 :)

**Review by daniel-wer — APPROVED** [review]
LGTM, so many bug fixes and improvements in one PR - great work :rainbow:

**Pull request merged**